tmt/convert.py

```diff
@@ -448,10 +448,11 @@ def read(
             restraint_file = True
             echo(style('Restraint ', fg='blue'), nl=False)
 
-    if filename is None:
+    if filename is None and not polarion:
         raise GeneralError('filename is not defined')
     # Open the datafile
     if restraint_file or makefile_file:
+        assert filename is not None  # type check
```

Comment:
Suspicious. The condition above would lead to `raise` if `filename` is false-ish and `polarion` isn't, but, can it happen that `filename` is false-ish and `polarion` is too? Then the condition above would not trigger, and therefore mypy cannot rule out the possibility of `filename` being `None` at this place. Would this be possible?

Reply:
I guess not, then it should hit other checks above and throw exception.